### PR TITLE
Adding CatBoost to the list of models

### PR DESCRIPTION
**Pull request by vishnuchalla**
## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] Optimization
- [x] Documentation Update

## Description
Adding CatBoost to the list of models as it one of well performing Gradient Boost based model that works well with categorical data.

## Related Tickets & Documents
Adding this as per the suggestion in project design document: https://docs.google.com/document/d/1CKAwKmOIumluvVUo8Qd224d5OGq-LSCHTu-nTs0Gy34/edit?disco=AAABpRINM2o

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
Tested and verified in local on CPU with around 8k records of sample data.
```
2025-10-03 18:48:06,394 - __main__ - INFO - Model Performance Summary:
2025-10-03 18:48:06,397 - __main__ - INFO - 
       Model  Overall R²  Overall RMSE  Overall MAE  Explained Variance
RandomForest    0.938880      0.235042     0.113551            0.938970
     XGBoost    0.934250      0.245289     0.122311            0.934259
    CatBoost    0.921117      0.268978     0.150960            0.921137
    LightGBM    0.898202      0.303874     0.157239            0.898221
2025-10-03 18:48:07,081 - src.multi_output_model - INFO - Model comparison plot saved to results/plots/model_comparison.png
2025-10-03 18:48:07,082 - __main__ - INFO - Generating feature importance for best model: RandomForest
2025-10-03 18:48:07,110 - __main__ - INFO - Top 10 most important features for RandomForest:
2025-10-03 18:48:07,111 - __main__ - INFO - 
                                 feature  importance
                         infraNodeMemory    0.244057
                 jobConfig.jobIterations    0.202203
                              ocpVersion    0.156950
                              k8sVersion    0.142383
                         masterNodesType    0.051182
                jobConfig.maxWaitTimeout    0.043661
                          infraNodesType    0.031628
                               ipsecMode    0.028817
podLatencyQuantilesMeasurement_Ready_P99    0.023953
                              totalNodes    0.015254
2025-10-03 18:48:07,111 - __main__ - INFO - Training pipeline completed successfully!
2025-10-03 18:48:07,111 - __main__ - INFO - Results saved to results/
2025-10-03 18:48:07,111 - __main__ - INFO - Models saved to models/
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,11 +147,11 @@ After training, find results in:
...
-      Model  Overall R²  Overall RMSE  Overall MAE  Explained Variance
- RandomForest      0.85          0.12         0.08               0.86
-     XGBoost      0.83          0.13         0.09               0.84
-     LightGBM      0.81          0.14         0.10               0.82
+Model,Overall R²,Overall RMSE,Overall MAE,Explained Variance
+XGBoost,-0.7409888020355209,0.42825588713063517,0.2727593458247816,-0.511705710492124
+CatBoost,-1.2120877876776353,0.4772675277039762,0.3252635852420828,-0.8362455427741021
+RandomForest,-1.2217242573256,0.39020471475639545,0.2541226539098206,-0.7973924742351571
+LightGBM,-106.36625193244984,0.673707178614463,0.5812945018216662,-6.167905692361981e-17
 ```
```
Comment (natoscott) [inline README.md:155]: @vishnuchalla I guess this is the first results of models trained with the increased dataset?  (the one with several thousand rows, and different features/targets).

Across the board the results look quite concerning.  Negative R-squared and Explained Variance values for all models suggests that none of the models are performing better than the null hypothesis (i.e. results are worse than random guess predictions).

One thing to try would be to start with a single target variable rather than many, and observe the results.  If the models predictions improve into a more acceptable range, start adding additional targets.

**Comment by vishnuchalla** [comment]
@natoscott @bentito @prasadovhal @VladimirKadlec ^^

**Comment by natoscott** [comment]
@vishnuchalla I added some review notes earlier.

**Comment by vishnuchalla** [comment]
> @vishnuchalla I added some review notes earlier.

Sorry, not sure what you are referring too. Can you please elaborate?

**Comment by natoscott** [comment]
> > @vishnuchalla I added some review notes earlier.
> 
> Sorry, not sure what you are referring too. Can you please elaborate?

Heh, my comments were in 'Pending' state - sorry - have posted 'em now.

File: README.md (hunk 1 above)
Comment (natoscott) [reply README.md:155]: Strangely, these negative R-squared and explained variance results in the README.md update don't match the results described in the PR description (which looks much better) ... whats happened there?

File: configs/config.yaml [hunk 2]
```diff
@@ -98,6 +98,16 @@ models:
       random_state: 42
       n_jobs: -1
 
+  - name: "CatBoost"
+    type: "CatBoostRegressor"
+    params:
+      iterations: 500
+      depth: 6
+      learning_rate: 0.05
+      loss_function: "RMSE"
```
Comment (natoscott) [inline configs/config.yaml]: @vishnuchalla based on this linked doc, it may be worthwhile trying "MultiRMSE" here:
https://catboost.ai/docs/en/concepts/loss-functions-multiregression#MultiRMSE

Comment (natoscott) [reply configs/config.yaml]: There may also be additional performance gains to be made for XGBoost using the (tree_method="hist", multi_strategy="multi_output_tree") parameters, which are for multi-output models like we have here:
https://xgboost.readthedocs.io/en/stable/tutorials/multioutput.html

**Review by VladimirKadlec — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (vishnuchalla) [reply README.md:155]: Sorry, I forgot to update the README.md earlier, did it now. Should address all your concerns.

File: configs/config.yaml (hunk 2 above)
Comment (vishnuchalla) [reply configs/config.yaml]: Added these suggestions and re-trained. Seems to have significance in long run as we have more data flowing in.

File: README.md (hunk 1 above)
Comment (natoscott) [reply README.md:155]: That looks better :)

**Review by natoscott — APPROVED** [review]
LGTM, thanks for the updates @vishnuchalla
